Structure_Engine/Create/Elements/RetainingWall.cs

```diff
@@ -0,0 +1,130 @@
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using BH.oM.Base.Attributes;
+using BH.oM.Geometry;
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.SurfaceProperties;
+using BH.Engine.Base;
+using BH.Engine.Geometry;
+using BH.Engine.Spatial;
+
+
+namespace BH.Engine.Structure
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates a RetainingWall from a Stem, Footing and properties.")]
+        [Input("stem", "The Stem of the RetainingWall.")]
+        [Input("footing", "The Footing of the RetainingWall.")]
+        [Input("retainedHeight", "The retained height of soil measured from the bottom of the wall Footing.")]
+        [Input("coverDepth", "The distance from top of Footing to finished floor level on the exposed face.")]
+        [Input("groundWaterDepth", "The distance from the base of the Footing to ground water level.")]
+        [Input("retentionAngle", "A property of the material being retained measured from the horizontal plane.")]
+        [Output("retainingWall", "The created RetainingWall containing the stem and footing.")]
+        public static RetainingWall RetainingWall(Stem stem, PadFoundation footing, double retainedHeight, double coverDepth, double retentionAngle, double groundWaterDepth = 0)
+        {
+            List<ICurve> curves = new List<ICurve>();
+            foreach (ICurve curve in stem.Outline.SplitAtPoints(stem.Outline.DiscontinuityPoints()))
+            {
+                Point start = curve.IStartPoint();
+                Point end = curve.IEndPoint();
+                curves.Add(curve);
+            }
+            ICurve bottomCurve = curves.OrderBy(p => p.IStartPoint().Z + p.IEndPoint().Z).First();
+
+            if (!footing.TopOutline.IsContaining(bottomCurve))
+            {
+                Base.Compute.RecordError("The stem is not connected to the footing. Make sure the stem bottom is on the footing outline.");
+                return null;
+            }
```

Comment:
As with the other method, I would put this in an `IsValid` method.

Reply:
Added IsValid method taking a stem and a footing and implementing it here.